### PR TITLE
Update build for Android 10 / Android Studio 4.0 Canary 5 + Day/Night themes

### PR DESCRIPTION
**Pull request by fat-tire**
It has been some time since Haven was updated, and with the release of Android 10 I figured why not bring it up to date.  I had some free time, and while I probably should have been reading Snowden's new book instead, this PR is the result.

There is now both a day and night theme that needs testing.  Hopefully theme changes won't affect functionality (especially if the change occurs while Haven is armed) but if this is an issue, I'm sure it wouldn't be hard to fix.  The dark theming colors were sort of picked by me just so it didn't look horrific.  Would appreciate subsequent refinement by UI experts since I don't know what I'm doing there.

Short list of changes:

* Update libraries
* Update gradle & gradle plugins
* Use daynight material themes to support dark mode
* New colors for dark mode
* Changes to ListActivity.java to keep overflow icon white regardless of theme
* CameraViewHolder.java changes for updated library & slight optimizations
* The logged images and stuff are now rooted at this.context.[getExternalFilesDir](https://developer.android.com/reference/android/content/Context.html#getExternalFilesDir(java.lang.String))(null), instead of deprecated Environment.[getExternalStorageDirectory](https://developer.android.com/reference/android/os/Environment#getExternalStorageDirectory())().  This is probably the most dramatic change in functionality, but something like this is necessary due to new storage permission policies in Android 10
* Add `attrs.xml` to address missing attributes in recyclerview library (temporary I hope)

If you build from the command line with `./gradlew assembleRelease` it should build fine out of the box.  Using the Android Studio UI might require commenting out `distributionSha256Sum` in `gradle-wrapper.properties` due to a "not fully supported to define distributionSha256Sum" error in newer Android Studio.  No big deal.

This PR is meant as a starting point for kickstarting some fresh development as no doubt libraries and plugins will continue to be updated, but just thought a refresh might be neato.

That's it!  Please test & enjoy.

ft

### CODE REVIEW

**Comment by fat-tire** [comment]
CI error is out of my hands :)

```
FAILURE: Build failed with an exception.
* What went wrong:
A problem occurred configuring root project 'haven'.
> Failed to install the following Android SDK packages as some licences have not been accepted.
     build-tools;29.0.2 Android SDK Build-Tools 29.0.2
     platforms;android-29 Android SDK Platform 29
  To build this project, accept the SDK license agreements and install the missing components using the Android Studio SDK Manager.
```

File: src/main/java/org/havenapp/main/PreferenceManager.java [hunk 1]
```diff
@@ -152,7 +150,6 @@ public void setSignalUsername (String signalUsername)
      *
      * @return the verified Signal username; null when no Signal username is verified even though registered.
      */
-    @Nullable
```
Comment (archie94) [inline src/main/java/org/havenapp/main/PreferenceManager.java]: Hmm.. Is there any reason as to why we are removing this annotation?

Comment (fat-tire) [reply src/main/java/org/havenapp/main/PreferenceManager.java]: Hmm.  Can't remember now.  Must have thrown a lint error is my best guess (?)  I'll probably update this PR soon and take a look at that.

Comment (fat-tire) [reply src/main/java/org/havenapp/main/PreferenceManager.java]: Update reverted this change in https://github.com/guardianproject/haven/pull/405/commits/a857b347621772c1991bfb0a7f5901a9b0f3df37

**Comment by n8fr8** [comment]
Amazing... as for the CI, yes, annoying. Working to update this to a Gitlab based CI soon.

Otherwise, reviewing, will test and merge soon.

**Comment by fat-tire** [comment]
Updated for 4.0 Canary 5, released today.  Also returned a @Nullability that no longer blocks the build (must have been fixed in gradle (?))  Anyway, see recent commits.

**Comment by n8fr8** [comment]
Just rolled back gradle tools version so I can still build this in my stable Android Studio 3.5.x. Otherwise, thanks for the swift kickstart in the rear :)

My kids are demanding I update Haven so we can catch Santa in the act this year, so definitely time for a new release.

**Pull request merged**